h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderHelper.java

```diff
@@ -82,6 +73,90 @@ static double computePriorMean(Frame encodings, int targetClass) {
     }
   }
 
+  /**
+   * @return the index of the interaction column for the group, or the index of the column if the group has only one.
+   */
+  static int createFeatureInteraction(Frame fr, String[] colGroup) {
+    return createFeatureInteraction(fr, colGroup, null);
+  }
+
+  /**
+   * @param interactionDomain the domain of the generated interaction column, if already known, 
+   *                          for example when computing interaction for predictions.
+   * @return the index of the interaction column for the group, 
+   * or the index of the column if the group has only one, 
+   * or -1 if one column in the interaction group is missing.
+   */
+  static int createFeatureInteraction(Frame fr, String[] colGroup, String[] interactionDomain) {
```

Comment:
Knit: I don't like the idea that 80% of TE functionality is in TEHelper. There could easily be `TargetEncoderFeatureInteractions` class at least.

Reply:
yes, not false… I followed the original code structure, but I could extract all MRTasks here into a separate package private class.